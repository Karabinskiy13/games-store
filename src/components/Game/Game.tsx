import React from 'react';
import Link from 'next/link';
import Image from 'next/image';

import { Games } from '../../../types';
import { Button, GameItem } from '../../styles';

const Game = ({ game, showModal }: Games) => {
  const posterLink = `http://${game.fields.poster?.fields.file.url}`;
  const showDescription = `/games/${game.fields.name}`;

  return (
    <GameItem>
      <Image
        className="poster"
        src={posterLink}
        width={270}
        height={400}
        alt="poster"
        data-testId="poster"
        onClick={() => showModal(true)}
      />
      <div className="name">{game.fields.name}</div>
      <div data-testId="price">{game.fields.price}UAH</div>
      <div data-testId="releaseDate">Release Date: {game.fields.releaseDate}</div>
      <Link href={showDescription}>
        <Button data-testId="button" className="button">
          {game.fields.showFull}
        </Button>
      </Link>
    </GameItem>
  );
};

export default Game;
