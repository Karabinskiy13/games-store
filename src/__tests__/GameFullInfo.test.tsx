import React from 'react';
import { jest } from '@jest/globals';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { IGames } from '../../contentful';
import Game from '../components/Game/Game';
import GamesInfo from '../pages/games/[id]';

const singleGame: any = {
  id: 1,
  quantity: 10,
  prevState: null,
  fields: {
    name: 'name',
    description: 'description',
    price: 100,
    releaseDate: '2020',
    platform: 'PS4',
    genres: 'action',
    fullDescription: 'full description',
    backdrop: {
      fields: {
        file: {
          url: 'https://images.unsplash.com/photo-1500309447817-512216363500?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=1050&q=80'
        }
      }
    }
  }
};

describe('<GameFullInfo>', () => {
  test('Should render component', () => {
    const { asFragment } = render(<GamesInfo gamesInfo={singleGame} />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('Should display poster', async () => {
    render(<GamesInfo gamesInfo={singleGame} />);
    await screen.findByTestId('wrapper');
    expect(screen.getByTestId('wrapper')).toBeInTheDocument();
  });

  test('Should render title of game', async () => {
    render(<GamesInfo gamesInfo={singleGame} />);
    await screen.findByText('name');
    expect(screen.getByText('name')).toBeInTheDocument();
  });

  test('Should render description of game', async () => {
    render(<GamesInfo gamesInfo={singleGame} />);
    await screen.findByText('full description');
    expect(screen.getByText('full description')).toBeInTheDocument();
  });

  test('Should render platform of game', async () => {
    render(<GamesInfo gamesInfo={singleGame} />);
    await screen.findByText('Aviable in PS4');
    expect(screen.getByText('Aviable in PS4')).toBeInTheDocument();
  });

  test('Should render price of game', async () => {
    render(<GamesInfo gamesInfo={singleGame} />);
    await screen.findByTestId('price');
    expect(screen.getByTestId('price')).toBeInTheDocument();
  });
});
