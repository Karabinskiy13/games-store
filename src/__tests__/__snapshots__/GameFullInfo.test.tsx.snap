// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GameFullInfo> Should render component 1`] = `
<DocumentFragment>
  <div>
    <div
      class="GameFull__Wrapper-sc-1n74d34-0 bbMpne"
      data-testid="wrapper"
      style="background-image: url(http://https://images.unsplash.com/photo-1500309447817-512216363500?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=1050&q=80);"
    >
      <div
        class="GameFull__Title-sc-1n74d34-1 kIFHkZ"
      >
        name
      </div>
      <div
        class="GameFull__Description-sc-1n74d34-2 hNuYfb"
      >
        full description
      </div>
      <div
        class="GameFull__Platform-sc-1n74d34-3 goeORr"
      >
        Aviable in PS4
      </div>
      <div
        class="GameFull__PriceButton-sc-1n74d34-5 fxsdUg"
      >
        <div
          class="GameFull__Price-sc-1n74d34-4 kXDvBh"
        >
          100 UAH
        </div>
        <button
          class="Games__Button-sc-itkhej-1 kRtork"
        >
          ADD TO CART
        </button>
      </div>
      <div>
        <input
          class="PurchaseModal__FeedBack-sc-wj1lb3-0 dSzayQ"
          placeholder="Comment"
          type="text"
        />
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
          style="max-width: 100px;"
          tabindex="0"
          type="button"
        >
          Add
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <div />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
